servicenow/stackstate_checks/servicenow/servicenow.py

```diff
@@ -272,16 +292,28 @@ def _process_relations(self, instance_info):
         """
         process relations
         """
-        relation_types = self._process_relation_types(instance_info)
         collected_relations = self._batch_collect(self._batch_collect_relations, instance_info)
         for relation in collected_relations:
-            parent_sys_id = relation['parent']['value']
-            child_sys_id = relation['child']['value']
-            type_sys_id = relation['type']['value']
-
-            relation_type = relation_types[type_sys_id]
-            data = self._filter_empty_metadata(relation)
-            data.update({"tags": instance_info.instance_tags})
+            try:
+                ci_relation = CIRelation(relation, strict=False)
+                ci_relation.validate()
+            except DataError as e:
+                self.log.warning("Error while processing properties - {}".format(e))
+                continue
+            data = {}
+            relation = self._apply_field_metadata(relation, instance_info.relation_display_value_list)
+            parent_sys_id = ci_relation.parent.value
+            child_sys_id = ci_relation.child.value
+            # first part after splitting with :: contains actual relation
+            relation_type = ci_relation.type.display_value.split("::")[0]
```

Comment:
Let's maybe add a check here to make sure the display value is not empty and that it contains a `::`, so something like:
```
if ci_relation.type.display_value and "::" in ci_relation.type.display_value
```

Reply:
it's actually a required field and we do that using schematics already. should we validate again?